internal/client/cluster/policy/policy_resource.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policyclusterclient
+
+import (
+	"net/url"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/helper"
+	policyclustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy/cluster"
+)
+
+// New creates a new cluster policy resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for cluster policy resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	ManageV1alpha1ClusterPolicyResourceServiceCreate(request *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyRequest) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse, error)
+
+	ManageV1alpha1ClusterPolicyResourceServiceDelete(fn *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyFullName) error
+
+	ManageV1alpha1ClusterPolicyResourceServiceGet(fn *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyFullName) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyGetPolicyResponse, error)
+
+	ManageV1alpha1ClusterPolicyResourceServiceUpdate(request *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyRequest) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse, error)
+}
+
+const (
```

Comment:
nit : constants should always be in the beginning of the file after import block